src/visitable.cpp

```diff
@@ -104,7 +104,7 @@ static int has_quality_internal( const T &self, const quality_id &qual, int leve
 
     self.visit_items( [&qual, level, &limit, &qty]( const item * e ) {
         if( e->get_quality( qual ) >= level ) {
-            qty = sum_no_wrap( qty, e->count_by_charges() ? int( e->charges ) : 1 );
+            qty = sum_no_wrap( qty, int( e->count() ) );
```

Comment:
Seems like `qty` and limit should actually be a long?

Reply:
I thought the same, but didn't want to suffer too much fix creep in this PR.  There are many other places in the code where charges is cast to an int; this is a can of worms I don't think is worth opening here.

Reply:
Fair enough, however I would then add a comment explaining we can probably overflow here and it needs to be fixed (even though it was actually this way before you changed anything).

Reply:
I opened #26103 to discuss the wider issue.  I really don't think it's worth a comment here, since it's so widespread.